src/traces/violin/attributes.js

```diff
@@ -0,0 +1,113 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var boxAttrs = require('../box/attributes');
+var scatterAttrs = require('../scatter/attributes');
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = {
+    y: boxAttrs.y,
+    x: boxAttrs.x,
+    x0: boxAttrs.x0,
+    y0: boxAttrs.y0,
+    name: boxAttrs.name,
+    orientation: extendFlat({}, boxAttrs.orientation, {
+        description: [
+            'Sets the orientation of the violin(s).',
+            'If *v* (*h*), the distribution is visualized along',
+            'the vertical (horizontal).'
+        ].join(' ')
+    }),
+
+    bandwidth: {
+        valType: 'number',
+        min: 0,
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Sets the bandwidth used to compute the kernel density estimate.',
+            'By default, the bandwidth is determined by Silverman\'s rule of thumb.'
+        ].join(' ')
+    },
+    scaleby: {
+        valType: 'enumerated',
+        values: ['width', 'area', 'count'],
+        dflt: 'width',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the method by which the width of each violin is determined.',
+            '*width* means each violin has the same (max) width',
+            '*area* means each violin has the same area',
+            '*count* means the violins are scaled by the number of sample points making',
+            'up each violin.'
+        ].join('')
+    },
+    span: {
+        valType: 'info_array',
+        items: [
+            {valType: 'any', editType: 'plot'},
+            {valType: 'any', editType: 'plot'}
+        ],
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Sets the span in data space for which the density function will be computed.',
+            'By default, the span goes from the minimum value to maximum value in the sample.'
+        ].join(' ')
+    },
+    side: {
```

Comment:
to remake, seaborn's 

![image](https://user-images.githubusercontent.com/6675409/31955618-ecd2f682-b8b6-11e7-8d40-4b2fa2b213dc.png)

Reply:
nice! also `'top'` and `'bottom'` for horizontal violins.

Reply:
As of https://github.com/plotly/plotly.js/pull/2116/commits/ad51966b2c72b59633415f711349b6836a9cf3e8, `side` is now an enumerated with values `'both'`, `'negative'` and `'positive'` so that the same set of values work with horizontal and vertical violins.